### ISSUE TITLE
ingredient-instruction-classifier crashing on Synology NAS install

### ISSUE THREAD

**Issue by jabib** [julianpoy/RecipeSage-selfhost#59]
*disclaimer: I still consider myself 'new' to Docker, so bear with me as much as you can allow here!*

I've managed to get the docker-compose.yaml up in Container Manager (the Synology implementation of Docker) and as a "project" it's created all 8 containers defined in the .yaml, however I'm running into an issue. The only thing I modified in the compose is the port assignment (37243:80)

Container ingredient-instruction-classifier keeps crashing. Logs just repeat:
```
Platform node has already been set. Overwriting the platform with node.
Illegal Instruction (core dumped)
```

I was having other issues but I _FINALLY_ figured out how to run the migration script and all else is right with the world!

**Comment by Evs91** [comment]
What type of CPU does your Synology have? If you don't know you can perhaps respond with the model of your Synology? It could be that your CPU (assuming it is not an Intel CPU but rather one of the RealTek or Marvell CPUs) is not compatible with the container due to tensorflow requiring certain instructions to be available on the CPU. Its a guess at this point without me confirming that.

**Comment by jabib** [comment]
DS920+ / Intel Celeron J4125

**Comment by jabib** [comment]
DS920+ / Intel Celeron J4125

**Comment by Evs91** [comment]
Without being able to really reproduce this due to me not owning or having access to Synology devices please take this with a grain of salt: 
     The instruction classifier is built upon a javascript implementation of Tensorflow which has the same requirements (best I can tell) as the standard implementation for Python (at least for CPU builds) that require AVX/AVX2 instructions to be supported on your processor. Celeron processors are at the "low end" of Intel CPUs and tend to not have features enabled on them. So while most Pentium processors have had AVX/AVX2 since the early 2010s, only recently (read: 2019/2020) have Celeron models gained these instruction sets. The [J4125 ](https://ark.intel.com/content/www/us/en/ark/products/197305/intel-celeron-processor-j4125-4m-cache-up-to-2-70-ghz.html)is not one listed as having AVX instruction compatibility. 
     The solution for you would be to build a custom container for tfjs and[ build the instruction classifier](https://github.com/tensorflow/tfjs/tree/master/tfjs-node#optional-build-optimal-tensorflow-from-source) with that custom version of tfjs that doesn't require the AVX instructions. 
    TL;DR: the default instruction classifier container is built on a PC that supports AVX instructions. You need to make one that doesn't require AVX. It's probably out of scope to promise to do one as modern CPUs (including 2021+ Celerons) all have the AVX instruction set.

**Comment by julianpoy** [comment]
@Evs91 is correct - this error indicates your CPU does not support AVX instructions.

See my comment here (and the comment linked from there): https://github.com/julianpoy/RecipeSage-selfhost/issues/41#issuecomment-1478594526
